### ISSUE TITLE
Add Train/Test Separation to convolutional_vae example

### ISSUE THREAD

**Issue by jramapuram** [blei-lab/edward#104]
The current example only shows the train division. To complete the test division scoping needs to be added. I.e. `phase=pt.Phase.train` or `phase=pt.Phase.test` . The way it is currently setup is incorrect as the testing is evaluated with the train phase. This causes the following issues:
  - Dropout is evaluated on test and train
  - batch_normalization is applied on test and train the same way (test needs to not update the running mean/var)

**Comment by mariru** [comment]
I think the train/test data split is required more generally. Not only for the convolutional_vae_example. We might consider adding an optional Data.test_data attribute to the Data class.

For evaluation we can then simply run Model.log_prob() on the test data.

**Comment by jramapuram** [comment]
Sure, that works as well. Just wanted to bring up the inconsistency with only having training logic and no testing logic!

**Comment by dustinvtran** [comment]
Sure that makes sense. Part of the reason I avoided this at the moment is due to the lack of a criticism interface. Criticism is all about assessing model fit with observed data, which in practice you do using data held-out from training time. So before we do any training/test split, we want to think hard about how to build the criticism interface following a principled probabilistic approach. Maja's pull request #105 makes sense as a first approach to this.
